chat-plugins/rps.js

```diff
@@ -0,0 +1,321 @@
+			'<button name="send" value="/rps choose R ' + this.gameId + '">Rock</button>' +
+			'<button name="send" value="/rps choose P ' + this.gameId + '">Paper</button>' +
+			'<button name="send" value="/rps choose S ' + this.gameId + '">Scissors</button></center><br><br>' +
+			"You have 60 seconds to make your choice.</center></div>";
+		player.send("|pm|~Rock/Paper/Scissors Host|" + player.userid + "|" + pmPost);
+	}
+
+	updateUsers() {
+		//get the latest user...
+		this.p1 = Users.get(this.p1.userid);
+		this.p2 = Users.get(this.p2.userid);
+	}
+
+	onChoose(user, choice) {
+		this.updateUsers();
+		if (user.userid !== this.p1.userid && user.userid !== this.p2.userid) return false;
+		let playerChoice = user.userid === this.p1.userid ? "p1choice" : "p2choice";
+		if (this[playerChoice]) return user.send("|pm|~Rock/Paper/Scissors Host|" + user.userid + "|/html You have already chosen your move!");
+		this[playerChoice] = choice;
+		user.send("|pm|~Rock/Paper/Scissors Host|" + user.userid + "|/html You have chosen: " + choiceNames[choice] + ".");
+		if (this.p1choice && this.p2choice) this.onEnd();
+	}
+
+	onEnd(inactivity) {
+		clearTimeout(this.timer);
+		// in the case of inactivity
+		if (inactivity) {
+			//determine winner
+			if (this.p1choice && !this.p2choice) {
+				this.p2.send("|pm|~Rock/Paper/Scissors Host|" + this.p2.userid + "|/html You have lost due to inactivity.");
+				this.parseWin(this.p1, this.p2, true);
+			} else if (!this.p1choice && this.p2choice) {
+				this.p1.send("|pm|~Rock/Paper/Scissors Host|" + this.p1.userid + "|/html You have lost due to inactivity.");
+				this.parseWin(this.p2, this.p1, true);
+			} else {
+				this.p1.send("|pm|~Rock/Paper/Scissors Host|" + this.p1.userid + "|/html You have lost due to inactivity.");
+				this.p2.send("|pm|~Rock/Paper/Scissors Host|" + this.p2.userid + "|/html You have lost due to inactivity.");
+			}
+			this.p1.RPSgame = null;
+			this.p2.RPSgame = null;
+			delete Rooms.global.RPS.games[this.gameId];
+			return;
+		}
+		let resultTable = {
+			"rr": "pp",
+			"rp": "p2",
+			"rs": "p1",
+			"pp": "pp",
+			"pr": "p1",
+			"ps": "p2",
+			"ss": "pp",
+			"sp": "p1",
+			"sr": "p2",
+		};
+		let winner, loser;
+		let gameResult = resultTable[this.p1choice.toLowerCase() + this.p2choice.toLowerCase()];
+		if (gameResult === "pp") {
+			//tie
+			this.p1.send("|pm|~Rock/Paper/Scissors Host|" + this.p1.userid + "|/html The game with " + this.p2.name + " was a tie! " + this.p2.name + " has chosen " + choiceNames[this.p2choice] + ".");
+			this.p2.send("|pm|~Rock/Paper/Scissors Host|" + this.p2.userid + "|/html The game with " + this.p1.name + " was a tie! " + this.p1.name + " has chosen " + choiceNames[this.p1choice] + ".");
+			if (this.gameType === "bucks") {
+				//return their 3 bucks each
+				Db('money').set(this.p1.userid, Db('money').get(this.p1.userid, 0) + 3);
+				Db('money').set(this.p2.userid, Db('money').get(this.p2.userid, 0) + 3);
+			}
```

Comment:
Pointing out that this script uses OriginDB, in which the economy is run through another source.